src/subdomains/core/history/mappers/transaction-dto.mapper.ts

```diff
@@ -49,12 +50,31 @@ export class TransactionDtoMapper {
       outputBlockchain: buyCrypto.target.asset.blockchain,
       outputPaymentMethod: CryptoPaymentMethod.CRYPTO,
       feeAmount: buyCrypto.totalFeeAmount
-        ? Util.roundReadable(
-            (buyCrypto.totalFeeAmount / buyCrypto.inputReferenceAmount) * buyCrypto.inputAmount,
-            isFiat(buyCrypto.inputAssetEntity),
-          )
+        ? Util.roundReadable(buyCrypto.totalFeeAmount * referencePrice, isFiat(buyCrypto.inputAssetEntity))
         : null,
       feeAsset: buyCrypto.totalFeeAmount ? buyCrypto.inputAsset : null,
+      fees: {
+        rate: buyCrypto.percentFee,
+        fixed:
+          buyCrypto.absoluteFeeAmount &&
+          Util.roundReadable(buyCrypto.absoluteFeeAmount * referencePrice, isFiat(buyCrypto.inputAssetEntity)),
+        min:
+          buyCrypto.minFeeAmount &&
+          Util.roundReadable(buyCrypto.minFeeAmount * referencePrice, isFiat(buyCrypto.inputAssetEntity)),
+        network:
+          buyCrypto.blockchainFee &&
+          Util.roundReadable(buyCrypto.blockchainFee * referencePrice, isFiat(buyCrypto.inputAssetEntity)),
+        dfx:
+          buyCrypto.totalFeeAmount &&
+          buyCrypto.blockchainFee &&
+          Util.roundReadable(
+            (buyCrypto.totalFeeAmount - buyCrypto.blockchainFee) * referencePrice,
+            isFiat(buyCrypto.inputAssetEntity),
+          ),
+        total:
+          buyCrypto.totalFeeAmount &&
+          Util.roundReadable(buyCrypto.totalFeeAmount * referencePrice, isFiat(buyCrypto.inputAssetEntity)),
+      },
```

Comment:
This could be done in a helper method and combined with the buy-fiat?
```ts
const referencePrice = buyFiat.inputAmount / buyFiat.inputReferenceAmount;
const fees = this.mapFees(buyFiat);

const dto: TransactionDto = {
   ...
   feeAmount: fees.total,
   fees,
   ...
};
```

```ts
private mapFees(entity: BuyCrypto | BuyFiat, referencePrice: number): FeeDto {
   ...
}
```